src/BenchmarkManager.py

```diff
@@ -38,6 +38,60 @@
 sns.color_palette()
 
 
+def _import_class(module_path, class_name, base_dir=None):
+    """
+    Helper function which allows to replace hard coded imports of the form
+    'import MyClass from path.to.mypkg' by calling _import_class('path.to.mypkg', 'MyClass').
+    If base_dir is specified its value will be added to python search path
+    if not not already contained in it.
+    :param module_path: python module path of the module containing the class to be imported
+    :type module_path: str
+    :param class_name: the name of the class to be imported
+    :type class_name: str
+    :rtype: classobj
+    """
+
+    #make sure that base_dir is in the search path. Otherwise
+    #the module imported here might not find its libraries.    
+    if not base_dir is None and not base_dir in sys.path:
+        logging.info(f"append to sys.path: {base_dir}")
+        sys.path.append(base_dir)
+    logging.info(f"import module {module_path}")    
+    module = importlib.import_module(module_path)
+    return vars(module)[class_name]    
+
+def _getInstanceWithSubOptions(options, class_name, *args):
+    """
+    Create an instance of the QUARK module (application, mapping, solver, device) identified by
+    class_name.
+    :param options: the section of the QUARK module configuration which is relevant here including the sub modules information.
+    :type options: list of dict
+    :param class_name: the name of the class to be initialized.
+    :type class_name: str
+    :param args: the list of arguments used for to the class initialization
+    """
+    for opt in options:
+        if not class_name == opt["name"]:
+            continue
+        clazz = _import_class(opt["module"], class_name, opt.get("dir"))
+        sub_options = None
+        for key in ["mappings","solvers","devices"]:
+            if key in opt:
+                sub_options = opt[key]
+                break
+        instance = clazz(*args)
+        
+        #sub_options inherits 'dir'
+        if not sub_options is None and "dir" in opt:
+            for sub_opt in sub_options:
+                if not "dir" in sub_opt:
+                    sub_opt["dir"] = opt["dir"]
+
+        instance.sub_options = sub_options
+        return instance
+    logging.warn(f"{class_name} not found in {options}")
```

Comment:
The 'warn' function is deprecated, use 'warning' instead .

```suggestion
    logging.warning(f"{class_name} not found in {options}")
```